tests/integration/Infrastructure/DashboardBundle/Validator/Constraints/ValidClientIdValidatorTest.php

```diff
@@ -76,4 +76,18 @@ public function test_invalid_client_id_url()
             'Expected certain violation but dit not receive it.'
         );
     }
+
+    public function test_invalid_client_id_containing_a_port()
```

Comment:
Nice test coverage! Consider adding a dataProvider here to test multiple client Id's in one go. That way you can test a multitude of illegal clientIds.

Also, what happens when a non numeric port 'number' is specified? Is that also covered by the UrlParser? Say: `https://foobar.example.com:illegal`